### PR TITLE
Add a method to generate all groups, on given configurations, selected by shadow schedulers

### LINKED ISSUE #334: Add a method to gather all manifests selected by a shadow scheduler
We have `generate_all_shadow_scheduler_tasks`, but we need an equivalent for manifests.

### PR DESCRIPTION
**Pull request by marco-c**
Fixes #334

### CODE REVIEW

**Comment by ahal** [comment]
Travis failures look legit (likely just need to update the tests)

File: mozci/task.py [hunk 1]
```diff
@@ -62,6 +62,18 @@ class Status(Enum):
 )
 
 
+def get_configuration_from_label(label: str) -> str:
```
Comment (ahal) [inline mozci/task.py:65]: Maybe you can refer to bug 1632870 here so if we end up fixing that and stumble across this comment, we can create an issue to switch over.

File: mozci/push.py [hunk 2]
```diff
@@ -1002,14 +1011,35 @@ def generate_all_shadow_scheduler_tasks(
...
+        for name in sorted(self.shadow_scheduler_names):
             try:
-                yield name, self.get_shadow_scheduler_tasks(name)
+                yield name, set(
+                    t["label"] for t in self.get_shadow_scheduler_tasks(name)
+                )
+            except Exception as e:
+                yield name, e
+
+    def generate_all_shadow_scheduler_config_groups(
```
Comment (ahal) [inline mozci/push.py:1022]: Won't force you to block on it, but would be nice to have a test similar to the task version:
https://github.com/mozilla/mozci/blob/master/tests/test_push.py#L361

**Review by ahal — APPROVED** [review]
Looks good, please fix the test failures first (and consider adding a new test).

File: mozci/push.py (hunk 2 above)
Comment (marco-c) [reply mozci/push.py:1022]: Done!

File: mozci/task.py (hunk 1 above)
Comment (marco-c) [reply mozci/task.py:66]: Done!

**Pull request merged**